datafusion/sqllogictest/test_files/sort_merge_join.slt

```diff
@@ -134,7 +134,6 @@ Alice 100 NULL NULL
 Alice 50 Alice 2
 Bob 1 NULL NULL
 NULL NULL Alice 1
-NULL NULL Alice 2
```

Comment:
This is incorrect. `Alice 2` was joined with `Alice 50` above. We should not produce null join row for it.

Reply:
double checked the joins, it looks correct